fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java

```diff
@@ -32,15 +32,15 @@
 public class DefaultPaymentPeriodsInOneYearCalculator implements PaymentPeriodsInOneYearCalculator {
 
     @Override
-    public Integer calculate(final PeriodFrequencyType repaymentFrequencyType) {
+    public Integer calculate(final PeriodFrequencyType repaymentFrequencyType,final PeriodFrequencyType interestRateFrequencyMethod) {
 
         Integer paymentPeriodsInOneYear = 0;
         switch (repaymentFrequencyType) {
             case DAYS:
-                paymentPeriodsInOneYear = 365;
+                paymentPeriodsInOneYear = interestRateFrequencyMethod.isMonthly() ? Integer.valueOf(360) : Integer.valueOf(365);
             break;
             case WEEKS:
-                paymentPeriodsInOneYear = 52;
+                paymentPeriodsInOneYear = interestRateFrequencyMethod.isMonthly() ? Integer.valueOf(48) : Integer.valueOf(52);
```

Comment:
This looks weird to me. If why would it be 48 if it is monthly?

Reply:
If the interest frequency is monthly, it looks more reasonable to assume there are four weeks in a month, so with 4 weeks in a month and 12 months in a year we shall have 4 x 12 = 48 weeks in a year.

Reply:
I dont think thats right... monthly interest calculation cannot be 4 weeks... @bharathc27 Thoughts?